pyro/distributions/planar.py

```diff
@@ -52,24 +52,28 @@ class PlanarFlow(Transform):
     codomain = constraints.real
 
     def __init__(self, input_dim):
-        super(PlanarFlow, self).__init__()
+        Transform.__init__(self)
+        nn.Module.__init__(self)
+
         self.input_dim = input_dim
-        self.module = nn.Module()
-        self.module.lin = nn.Linear(input_dim, 1)
-        self.module.u = nn.Parameter(torch.Tensor(input_dim))
+        self.lin = nn.Linear(input_dim, 1)
+        self.u = nn.Parameter(torch.Tensor(input_dim))
         self.reset_parameters()
         self._intermediates_cache = {}
         self.add_inverse_to_cache = True
 
     def reset_parameters(self):
-        stdv = 1. / math.sqrt(self.module.u.size(0))
-        self.module.lin.weight.data.uniform_(-stdv, stdv)
-        self.module.u.data.uniform_(-stdv, stdv)
+        stdv = 1. / math.sqrt(self.u.size(0))
+        self.lin.weight.data.uniform_(-stdv, stdv)
+        self.u.data.uniform_(-stdv, stdv)
+
+    def __hash__(self):
```

Comment:
:+1: @stefanwebb I'm ok with any pattern you prefer in this PR. Could you also submit a Pyro PR that implements what you believe to be the right solution? I'd like to discuss your solution in that PR, and if everything works, later move it upstream in a PyTorch PR. Also feel free to directly send a PyTorch PR, but the review latency may be much higher.

Reply:
I.e. please also implement the right solution in pyro/distributions/torch.py somehow, either as a __hash__ method or a `TransformedDistributionModule` or whatever.